subprojects/android-test/ui-testing-core-app/src/androidTest/kotlin/com/avito/android/ui/test/KeyboardTest.kt

```diff
@@ -0,0 +1,230 @@
+package com.avito.android.ui.test
+
+import android.view.WindowManager
+import com.avito.android.rule.InHouseScenarioScreenRule
+import com.avito.android.test.Device
+import com.avito.android.ui.KeyboardActivity
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import ru.avito.util.assertThrows
+
+class KeyboardTest {
+
+    @get:Rule
+    val rule = object : InHouseScenarioScreenRule<KeyboardActivity>(KeyboardActivity::class.java) {}
+
+    @Before
+    fun before() {
+        rule.launchActivity(null)
+    }
+
+    @Test
+    fun openActivity_keyboardIsNotDisplayed_whenAdjustUnspecified() {
```

Comment:
IDK how to make these tests better without Parameterized mechanism

Reply:
The proposition is to make a few classes and split cases i.e. by Adjust type
And make a Base TestClass wherein constructor we will pass a type parameter

Reply:
Nice idea!

Reply:
Does it work? See MBS-10589

Reply:
Unfortunately, no. It doesn't work

Reply:
What about place test methods in the base class and override them to hack runner